node/src/components/consensus/cl_context.rs

```diff
@@ -43,7 +43,7 @@ impl ValidatorSecret for Keypair {
     type Signature = Signature;
 
     fn sign(&self, hash: &Digest) -> Signature {
-        crypto::sign(hash, self.secret_key.as_ref(), &self.public_key)
+        crypto::sign(hash, self.secret_key.as_ref(), &self.public_key).unwrap()
```

Comment:
Sadly, we can't panic here either.

These errors need to be surfaced all the way up to the `handle` method on our components, and we have to emit `fatal!(efffect_builder,...)` effects.